interfaces/apparmor/apparmor.go

```diff
@@ -117,9 +118,32 @@ func loadProfiles(fnames []string, cacheDir string, flags aaParserFlags) error {
 	if !osutil.GetenvBool("SNAPD_DEBUG") {
 		args = append(args, "--quiet")
 	}
-	args = append(args, fnames...)
 
-	output, err := exec.Command("apparmor_parser", args...).CombinedOutput()
+	parser, internal, err := apparmor_sandbox.FindAppArmorParser()
+	if err != nil || !internal {
+		// if we couldn't find the parser with apparmor_sandbox
+		// then fall-back to trying to find one in the current PATH
+		// - same for if we are not using the internal
+		// apparmor_parser so that we can support a mocked parser
+		// during tests
+		parser = "apparmor_parser"
+	} else {
+		// when using the internal apparmor_parser also use it's
+		// own configuration and includes etc plus also ensure we
+		// use the 3.0 feature ABI to get the widest array of
+		// policy features across the widest array of kernels
+		// versions
+		prefix := strings.TrimSuffix(parser, "apparmor_parser")
+		args = append(args, "--config-file")
+		args = append(args, filepath.Join(prefix, "/apparmor/parser.conf"))
+		args = append(args, "-b")
+		args = append(args, filepath.Join(prefix, "/apparmor.d"))
+		args = append(args, "--policy-features")
+		args = append(args, filepath.Join(prefix, "/apparmor.d/abi/3.0"))
```

Comment:
```suggestion
		args = append(args, []string{
			"--config-file", filepath.Join(prefix, "/apparmor/parser.conf"),
			"--base", filepath.Join(prefix, "/apparmor.d"),
			"--policy-features", filepath.Join(prefix, "/apparmor.d/abi/3.0"),
		}...)
```

Reply:
That looks much nicer - thanks - https://github.com/snapcore/snapd/pull/9939/commits/35acd6d3265a3aa7489c1e8e45735038807073bb